tests/validation/tests/v3_api/test_multi_cluster_app.py

```diff
@@ -296,103 +320,102 @@ def test_multi_cluster_cluster_answer_override():
+    client = get_user_client()
+    answer_values = get_defaut_question_answers(client, MYSQL_EXTERNALID_131)
+    mcapp = client.create_multiClusterApp(
+        templateVersionId=MYSQL_TEMPLATE_VID_131,
         targets=targets,
-        roles=["cluster-owner"],
+        roles=PROJECT_ROLE,
         name=random_name(),
         answers=[{"values": answer_values}],
-        upgradeStrategy=NEW_UPGRADE)
-    multiclusterapp = wait_for_mcapp_to_active(client, multiclusterapp)
-    project_client1 = project_detail["p_client0"]
-    project_client2 = project_detail["p_client1"]
-    app = multiclusterapp.targets[0].projectId.split(":")
-    app1id = app[1] + ":" + multiclusterapp.targets[0].appId
-    app2 = multiclusterapp.targets[1].projectId.split(":")
-    app2id = app2[1] + ":" + multiclusterapp.targets[1].appId
+        upgradeStrategy=ROLLING_UPGRADE_STRATEGY)
+    mcapp = wait_for_mcapp_to_active(client, mcapp)
     new_answers = {"values": answer_values}
     new_answers["values"]["mysqlUser"] = "admin1234"
-    multiclusterapp = client.update(multiclusterapp,
-                                    roles=["cluster-owner"],
-                                    answers=[new_answers])
-    multiclusterapp = client.reload(multiclusterapp)
+    mcapp = client.update(mcapp,
+                          roles=["cluster-owner"],
+                          answers=[new_answers])
+    mcapp = client.reload(mcapp)
+    app_info = {"p_client": None, "app_id": None}
+
     start = time.time()
-    upgraded = False
-    # assert apps have different states and answers
-    while time.time()-start < 30 or upgraded == False:
-        upgraded = return_application_status_and_upgrade(
-            project_client1, app1id, project_client2, app2id)
-        time.sleep(.1)
-    assert upgraded == True, "did not upgrade correctly"
-    time.sleep(20)
-    # since one has updated, asserts that both apps are in teh same state
-    while time.time()-start < 100 or upgraded == True:
-        upgraded = return_application_status_and_upgrade(
-            project_client1, app1id, project_client2, app2id)
-        time.sleep(.1)
-    assert upgraded == False, "did not upgrade correctly"
-    validate_multi_cluster_app_cluster(multiclusterapp)
-    delete_multi_cluster_app(multiclusterapp)
+    end = time.time()
+    time.sleep(5)
+    app_state = []
+    for i in range(0, len(mcapp.targets)):
+        app_id = mcapp.targets[i].appId
+        assert app_id is not None, "app_id is None"
+        project_client = project_detail["p_client" + str(i)]
+        app_detail = project_client.list_app(id=app_id).data[0]
+        app_state.append(app_detail.state)
+        if app_detail.state == "active":
+            app_info["p_client"] = project_client
+            app_info["app_id"] = app_id
+    assert app_state.count("active") == 1, "Only one app should be upgrading"
+    print("app_state: ", app_state)
+    # check interval time is 20 seconds
```

Comment:
where is the 20s sleeping? Do you mean sleeping 5s at line 388?

Reply:
skipping this test for now